code/modules/clothing/modular_armor/modular.dm

```diff
@@ -493,7 +505,9 @@
 	if(!new_color || !do_after(user, 1 SECONDS, TRUE, src, BUSY_ICON_GENERIC))
 		return
 
-	set_greyscale_colors(new_color)
+	if(new_color)
+		set_greyscale_colors(new_color)
```

Comment:
There is a check already on L505.

You'll likely want to add a check for a new hat colour being selected others selecting a hat colour and not an armour colour will not update the hate.

Reply:
yeah, i'll check some stuff, still need some code to do since thats from last year and this code around got updated